src/components/MapView/Analyser/index.tsx

```diff
@@ -187,8 +195,47 @@ function Analyser({ classes }: AnalyserProps) {
     };
 
     await dispatch(requestAndStoreAnalysis(params));
+    // eslint-disable-next-line fp/no-mutating-methods
+    history.replace(
+      `?hazardLayerId=${hazardLayerId}&baselineLayerId=${baselineLayerId}&date=${selectedDate}&statistic=${statistic}&aboveThreshold=${aboveThreshold}&belowThreshold=${belowThreshold}`,
```

Comment:
We need to find a better way than hard coding all the fields into the url. After talking with Eric, `useReducer` seems like a perfect fit here, as it will give you access to the object key names, so the url will automatically add new fields as they are introduced without any extra work. Let me know if you need help implementing this.

Reply:
@Ry-DS Do you mean to have a reducer just for the URL which updates on every input field update?